conductor_api/src/config.rs

```diff
@@ -351,6 +354,38 @@ impl Configuration {
 
         self
     }
+
+    /// This function checks if there is duplicated file storage from the instances section of a provided
+    /// TOML configuration file. For efficiency purposes, we short-circuit on the first encounter of a
+    /// duplicated values.
+    fn check_instances_storage(&self) -> Result<(), String> {
+        let storage_paths: Vec<&str> = self
+            .instances
+            .iter()
+            .filter_map(|stg_config| match stg_config.storage {
+                StorageConfiguration::File { ref path }
+                | StorageConfiguration::Pickle { ref path } => Some(path.as_str()),
+                _ => None,
+            })
+            .collect();
+
+        // Here we don't use the already implemented 'detect_dupes' function because we don't need
+        // to keep track of all the duplicated values of storage instances. But instead we use the
+        // return value of 'HashSet.insert()' conbined with the short-circuiting propiety of
+        // 'iter().all()' so we don't iterate on all the possible value once we found a duplicated
+        // storage entry.
+        let mut path_set: HashSet<&str> = HashSet::new();
+        let has_uniq_values = storage_paths.iter().all(|&x| path_set.insert(x));
+
+        if !has_uniq_values {
+            Err(String::from(
+                "Forbidden duplicated file storage value encountered.\
+                 Please fix this in the 'TOML Conductor configuration file'.",
```

Comment:
The second instance seems inappropriate as error message since the config does not necessarily come from a file. See comments above.

Reply:
You're probably right, thanks :+1: